### PR TITLE
Allow compilation of mgridgen (serial) as well as parmgridgen (parallel)

### PR DESCRIPTION
**Pull request by olesenm**

### CODE REVIEW

File: var/spack/repos/builtin/packages/parmgridgen/package.py [hunk 1]
```diff
@@ -26,46 +26,58 @@
...
-        install("MGridGen/IMlib/libIMlib.a",
-                join_path(prefix.lib, 'libIMlib.a'))
         install("libmgrid.a", prefix.lib)
-        install("libparmgrid.a", prefix.lib)
+        install("mgridgen",   prefix.bin)
+
+        # Can't do much without the corresponding headers
+        # install(
+        #     "MGridGen/IMlib/libIMlib.a",
+        #     join_path(prefix.lib, 'libIMlib.a'))
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/parmgridgen/package.py]: Is this TODO or should the commented code be removed?

Comment (olesenm) [inline var/spack/repos/builtin/packages/parmgridgen/package.py]: No problem - I left it in as being some random rubbish that I could figure out why it was there, but didn't want to delete it without asking first. Given your comment, it really does seem to random rubbish that I'll remove now.

**Comment by olesenm** [comment]
Any thoughts/opinions about renaming this `mgridgen` with an `+mpi` option?
This seems to fit better with packages like `scotch` that have both serial and parallel versions together.

**Comment by adamjstewart** [comment]
The serial version and the parallel version have two different tarballs, right?

**Comment by olesenm** [comment]
Same tarball, same make file (like scotch, unlike metis,parmetis).

**Comment by adamjstewart** [comment]
I've always sided with the developers and used whatever they decided to name the tarball to choose the package name. In this case, the developers say:

> Both the serial and the parallel library come in one package, namely PARMGRIDGEN, version 1.0.

So I vote for `parmgridgen`. But I don't use this package aside from as a dependency for things I actually care about, so my opinion isn't very strong.

**Comment by tgamblin** [comment]
I agree with @adamjstewart -- I'd go with one package called parmgridgen with an mpi variant.  But I don't feel too strongly about that and would be fine with mgridgen+mpi too.

**Comment by olesenm** [comment]
OK - I don't care too strongly either. So we'll leave it as parmgridgen, which means it should now be ready to merge.

**Comment by olesenm** [comment]
@tgamblin - It looks like you need to approve the changes so that this can get merged.

**Pull request merged**